setup.py

```diff
@@ -65,4 +65,10 @@
             "dev": doc_requirements + test_requirements,
         },
         zip_safe=False,
+        use_scm_version={
+            "write_to": "_version.py",
+            "write_to_template": '__version__ = "{version}"',
+            "tag_regex": r"^(?P<prefix>v)?(?P<version>[^\+]+)(?P<suffix>.*)?$",
```

Comment:
Can you walk me through what this is doing?

Reply:
The `use_scm_version` block in `setup.py` is taken from [an example in the documentation](https://github.com/pypa/setuptools_scm#configuration-parameters). `use_scm_version` says to use setuptools scm, `write_to` and `write_to_template` manages the `_version.py` file writing. The `tag_regex` matches tags of valid version names, used in discerning the current version. This regex could be paired back on retrospect, since we'll never release a version that has a `+` segment (PyPI disallows this AFAIK)